client/sites-dashboard/components/sites-dashboard-ctas.tsx

```diff
@@ -0,0 +1,38 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { useI18n } from '@wordpress/react-i18n';
+import { addQueryArgs } from 'calypso/lib/url';
+import {
+	CreateSiteIcon,
+	CREATE_SITE_ICON_SPARKLES,
+	MigrateSiteIcon,
+	MIGRATE_SITE_ICON_SPARKLES,
+} from './sites-dashboard-cta-icons';
+import { SparklingCTA } from './sparkling-cta';
+
+export const CreateSiteCTA = () => {
+	const { __ } = useI18n();
+
+	const newSiteURL = isEnabled( 'hosting-onboarding-i2' ) ? '/setup/new-hosted-site' : '/start';
```

Comment:
I think I have the same concern about _only_ using the feature flag to decide which site creation flow to use. When we launch this feature, `hosting-onboarding-i2` turns `true`, and all users will be seeing our `new-hosted-site` flow.

That means any regular user, who deletes all their sites to start again, if they click this button they'll only be offered the Business and Commerce plan and be taken straight to WoA. These users may just want to create a free blog.

I think we want this button to go to `/setup/new-hosted-site` only if (1) the feature is enabled, and (2) they've just come from the `/hosting` LOHP. Otherwise we have no hint that they're interested in only WoA sites.